TrafficCapture/captureOffloader/src/main/java/org/opensearch/migrations/trafficcapture/StreamChannelConnectionCaptureSerializer.java

```diff
@@ -280,6 +280,9 @@ private void addDataMessage(int captureFieldNumber, int dataFieldNumber, Instant
             bb = bb.slice();
             byteBuffer.position(byteBuffer.position() + chunkBytes);
             addSubstreamMessage(segmentFieldNumber, segmentDataFieldNumber, segmentCountFieldNumber, ++dataCount, timestamp, bb);
+            if (getOrCreateCodedOutputStream().spaceLeft() < (availableCOSSpace - chunkBytes - trafficStreamOverhead)) {
+                log.warn("Writing segment substream to CodedOutputStream took more space than max calculated, this should be investigated");
```

Comment:
If this is a programmer error, we should have a full assertion or thrown exception instead of a warning message.  This would be a critical error, blocking the current observations from being sent, right?

Reply:
There are cases where this could be a critical error and cases where it's not. In the cases where we are filling up the CodedOutputStream as much as we can and we didn't calculate enough size we will already hit a no space available in CodedOutputStream error as a result.  There are other cases where we could potentially write more than we thought but would be pretty much untraceable since there was also enough space in the CodedOutputStream to handle. This pattern could go undetected unless we got a message that was close to the bounds.

Because of this I don't want to throw an exception since we could very well handle it, and if not the CodedOutputStream will throw an exception. I've added another check like this to a normal message that doesn't need segmented

Reply:
Hmmmm - so you're saying that your overage will make this a softer landing in many cases.
OK - let me turn this around.  If a user sees this, how can it be investigated?  The log message doesn't give me any useful information.  Can you at least put the sizes, maybe some ids (in case there are additional previous logs)

Reply:
Sure good callout, I have added some additional details to this log line